DFC.App.FindACourse/Controllers/CourseController.cs

```diff
@@ -498,6 +505,26 @@ private static string NormalizePostcode(string postcode)
             return postcode.Insert(postcode.Length - 3, " ");
         }
 
+        private async Task<IActionResult> FilterResultsInternal(BodyViewModel model)
+        {
+            var newBodyViewModel = await GenerateModelAsync(model).ConfigureAwait(false);
+
+            try
+            {
+                model.Results = await findACourseService.GetFilteredData(newBodyViewModel.CourseSearchFilters, newBodyViewModel.CourseSearchOrderBy, model.RequestPage).ConfigureAwait(false);
+                model.UsingAutoSuggestedLocation = newBodyViewModel.UsingAutoSuggestedLocation;
+                model.SideBar.DidYouMeanLocations = newBodyViewModel.SideBar.DidYouMeanLocations;
+            }
+            catch (Exception ex)
+            {
+                logService.LogError($"{nameof(this.FilterResults)} threw an exception" + ex.Message);
+            }
+
+            logService.LogInformation($"{nameof(this.FilterResults)} generated the model and ready to pass to the view");
```

Comment:
When an exception occurs, this log message will also be recorded. Maybe it should be moved to after line 516 so it is only logged for when successful?